### PR TITLE
feat(client): Add `compute_output_root` to block executor

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a function to the block executor that allows for generating the current output root, based on the parent timestamp and current trie.

### CODE REVIEW

**Comment by clabby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/225?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#225** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/225?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#210** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/210?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#206** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/206?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#205** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/kona/205?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: bin/programs/client/src/l2/executor/mod.rs [hunk 1]
```diff
@@ -275,6 +275,49 @@ where
...
+    /// output_root = keccak256(version_byte .. payload)
+    /// payload = state_root .. withdrawal_storage_root .. latest_block_hash
+    /// ```
+    ///
+    /// ## Returns
+    /// - `Ok(output_root)`: The computed output root.
+    /// - `Err(_)`: If an error occurred while computing the output root.
+    pub fn compute_output_root(&mut self) -> Result<B256> {
+        const OUTPUT_ROOT_VERSION: u8 = 0;
+        const L2_TO_L1_MESSAGE_PASSER_ADDRESS: Address =
```
Comment (refcell) [inline bin/programs/client/src/l2/executor/mod.rs:292]: This should be grabbed from the superchain-registry no?

**Review by refcell — APPROVED** [review]
Looks good!

File: bin/programs/client/src/l2/executor/mod.rs (hunk 1 above)
Comment (clabby) [reply bin/programs/client/src/l2/executor/mod.rs:292]: Yeah good callout. Need some predeploy address constants in there, then can do! Will wait to merge, looks like there are some open PRs over there.

**Pull request merged**